tests/validators/test_int.py

```diff
@@ -135,6 +147,84 @@ def test_positive_int(input_value, expected):
         assert isinstance(output, int)
 
 
+@pytest.mark.parametrize(
+    'input_value,expected',
+    [
+        (-1, -1),
+        (0, Err('Input should be less than 0 [type=less_than, input_value=0, input_type=int]')),
+        (-i64_max, -i64_max),
+        (-i64_max - 1, -i64_max - 1),
+        (-int('9' * 30), -int('9' * 30)),
+    ],
+)
+def test_negative_int(input_value, expected):
+    v = SchemaValidator({'type': 'int', 'lt': 0})
+    if isinstance(expected, Err):
+        with pytest.raises(ValidationError, match=re.escape(expected.message)):
+            v.validate_python(input_value)
+    else:
+        output = v.validate_python(input_value)
+        assert output == expected
+        assert isinstance(output, int)
+
+
+@pytest.mark.parametrize(
+    'input_value,expected',
+    [
+        (1, 1),
+        (i64_max, i64_max),
+        (i64_max + 1, i64_max + 1),
+        (i64_max * 2, i64_max * 2),
+        (
+            int(1e30),
+            Err(
+                'Input integer too large to convert to 64-bit integer '
+                '[type=int_overflow, input_value=1e+30, input_type=float]'
+            ),
+        ),
+        (0, Err('Input should be greater than 0 [type=greater_than, input_value=0, input_type=int]')),
+        (-1, Err('Input should be greater than 0 [type=greater_than, input_value=-1, input_type=int]')),
+        pytest.param(
+            [1, 2],
+            Err('Input should be a valid integer [type=int_type, input_value=[1, 2], input_type=list]'),
+            id='list',
+        ),
+    ],
+)
+def test_positive_json(input_value, expected):
+    v = SchemaValidator({'type': 'int', 'gt': 0})
+    json_input = json.dumps(input_value)
+    if isinstance(expected, Err):
+        with pytest.raises(ValidationError, match=re.escape(expected.message)):
+            v.validate_json(json_input)
+    else:
+        output = v.validate_json(json_input)
+        assert output == expected
+        assert isinstance(output, int)
+
+
+@pytest.mark.parametrize(
+    'input_value,expected',
+    [
+        (-1, -1),
+        (0, Err('Input should be less than 0 [type=less_than, input_value=0, input_type=int]')),
+        (-i64_max, -i64_max),
+        (-i64_max - 1, -i64_max - 1),
+        (-i64_max * 2, Err('Input integer too large to convert to 64-bit integer [type=int_overflow, input_value=')),
```

Comment:
It would be nice if this was the warning shown for strings with too many digits

Reply:
done.